src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  return actions.reduce((acc, action, index) => {
+    const statePrevious = index === 0 ? { ...state } : { ...acc[index - 1] };
+
+    switch (action.type) {
+      case 'addProperties':
+        acc.push(Object.assign({}, statePrevious, action.extraData));
```

Comment:
According to the checklist, avoid repeating the same action in all `switch` cases. Since you are creating a copy of the state in each case, this should be done once after the switch block. This helps in adhering to the [DONT REPEAT YOURSELF] principle.